GoogleForCreators/web-stories-wp
# Design System Storybook - Buttons

## Summary

Adds the new design system's button component in rectangle and circle variants with option to have an icon as a button and a link as a button.

<img width="873" alt="Screen Shot 2020-10-26 at 9 35 47 AM" src="https://user-images.githubusercontent.com/10720454/97200621-af4f8e00-176e-11eb-96bb-0d32bd279c59.png">

## Relevant Technical Choices

- Tried to simplify the Button's API using the edit-story and dashboard's existing button use cases. 
- One `Button` component with exported constants to guide development with `BUTTON_VARIANTS` (rectangle as default, options are circle, icon), `BUTTON_SIZES` (small, medium), `BUTTON_TYPES` (mirror existing button types found in edit stories and dashboard - primary, secondary, tertiary, plain). 
- Will swap to a link if a `href` is present. 

## To-do

Question out to Sam regarding color differences between theme values and button values.
In[ figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=1286%3A456428) (bolded values are the colors set on the buttons that do not sync with theme colors): 
1) primary colors: #CBACFF (normal), **#B695EE** (hover, focus), **#EFEFE**/ #A1A09B (disabled)
2) Secondary colors: **#D1D1CC** (normal), #A1A09B (hover, focus), **#EFEFE**/ #A1A09B (disabled)
3) Tertiary colors:  #fff (normal), **#EFEFEF** (hover, focus), white/#A1A09B (disabled)

## User-facing changes

None, changes exist only in storybook

## Testing Instructions

- Run storybook on branch locally, see new design-system/components/button, see different button options. Mimics figma - right now the rectangle button has only the small size so medium loads the same but I wanted to loop through all the possible combos so I left it so that if anything changes it's easy to see. 

---



Addresses #4936
